controllers/cpms/cpms_controller.go

```diff
@@ -60,10 +62,34 @@ func (r *CPMSReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	}
 
 	reqLogger.Info("Found ControlPlaneMachineSet")
+
+	// Fetch the instance type from cpms spec
+	specRaw := cpms.Spec.Template.OpenShiftMachineV1Beta1Machine.Spec.ProviderSpec.Value.Raw
```

Comment:
Right now this is the only valid value (so all CPMS will be MachineV1Beta1), but this is a discriminated union type, so this *might* change in the future.

https://docs.openshift.com/container-platform/4.12/rest_api/machine_apis/controlplanemachineset-machine-openshift-io-v1.html#spec-template

If we want to future-proof this, we should check that the value in `machineType` is actually `machines_v1beta1_machine_openshift_io` before trying to access this field - and if it's not error out.